AsyncDisplayKit/Private/ASControlTargetAction.m

```diff
@@ -0,0 +1,66 @@
+//
+//  ASControlTargetAction.m
+//  AsyncDisplayKit
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+
+#import "ASControlTargetAction.h"
+
+@implementation ASControlTargetAction
+{
+  id _target;
+  BOOL _createdWithNoTarget;
+}
+
+- (void)setTarget:(id)target {
+  _target = target;
+  
+  if (!target) {
+    _createdWithNoTarget = YES;
+  }
+}
+
+- (id)target {
+  return _target;
+}
+
+- (BOOL)isEqual:(id)object {
+  if (![object isKindOfClass:[ASControlTargetAction class]]) {
```

Comment:
This seems pretty complex.  Can you remove some of this logic by starting out with ASObjectIsEqual() ?